csrc/executor_utils.cpp

```diff
@@ -407,11 +407,67 @@ void validateAlignedVectorizeExtents(
   //     info.word_size);
 }
 
+namespace {
+
+// Return offsets of the first points accessed. Currently only
+// non-zero when tensor is sliced.
+std::vector<size_t> getTensorOffsets(
+    TensorView* tv,
+    const std::vector<int64_t>& strides,
+    ExpressionEvaluator& eval) {
+  if (!tv->isFusionInput()) {
+    return {0};
+  }
+
+  std::vector<size_t> offsets;
```

Comment:
Should we use `std::unordered_set< size_t >`?

Reply:
Updated